test/expected/function_calls.out

```diff
@@ -1940,4 +1940,162 @@ begin;
 (1 row)
 
     set search_path to default;
+    rollback to savepoint a;
+    create function add_smallints(a smallint default 1, b smallint default 2)
+        returns smallint language sql immutable
+    as $$ select a + b; $$;
+    select jsonb_pretty(
+        graphql.resolve($$
+            query IntrospectionQuery {
+              __schema {
+                queryType {
+                  name
+                  fields {
+                    name
+                    args {
+                      name
+                      defaultValue
+                      type {
+                        name
+                      }
+                    }
+                  }
+                }
+              }
+            }
+        $$)
+    );
+                     jsonb_pretty                     
+------------------------------------------------------
+ {                                                   +
+     "data": {                                       +
+         "__schema": {                               +
+             "queryType": {                          +
+                 "name": "Query",                    +
+                 "fields": [                         +
+                     {                               +
+                         "args": [                   +
+                             {                       +
+                                 "name": "a",        +
+                                 "type": {           +
+                                     "name": "Int"   +
+                                 },                  +
+                                 "defaultValue": null+
```

Comment:
current the defaults do not render in the introspection schema
that's where IDEs like GraphiQL pull the info and do syntax highlighting from so we should support it there too

Reply:
The defaults can be extracted from the `proargdefaults` column of the `pg_proc` table. This column's type is `pg_node_tree` which can be tricky to parse as it looks like this: `({CONST :consttype 16 :consttypmod -1 :constcollid 0 :constlen 1 :constbyval true :constisnull false :location 132 :constvalue 1 [ 0 0 0 0 0 0 0 0 ]})`.

Calling `pg_get_expr` built-in function on a `pg_node_tree` type returns a string representation which looks more manageable but can still be quite involved for complex values like json & jsonb.

Some example values for `pg_get_expr(proargdefaults)` below:

```
postgres=# select proname, pronargs, pg_get_expr(proargdefaults, 0) from pg_catalog.pg_proc where proargdefaults is not null;
               proname               | pronargs |            pg_get_expr
-------------------------------------+----------+------------------------------------
 jsonb_insert                        |        4 | false
 jsonb_path_query                    |        4 | '{}'::jsonb, false
 jsonb_path_exists_tz                |        4 | '{}'::jsonb, false
 jsonb_path_query_array_tz           |        4 | '{}'::jsonb, false
 pg_logical_slot_peek_changes        |        4 | '{}'::text[]
 pg_logical_slot_peek_binary_changes |        4 | '{}'::text[]
 pg_create_logical_replication_slot  |        4 | false, false
 is_normalized                       |        2 | 'NFC'::text
 jsonb_set_lax                       |        5 | true, 'use_json_null'::text
 parse_ident                         |        2 | true
 pg_terminate_backend                |        2 | 0
 json_populate_record                |        3 | false
 json_populate_recordset             |        3 | false
 pg_logical_slot_get_changes         |        4 | '{}'::text[]
 pg_logical_slot_get_binary_changes  |        4 | '{}'::text[]
 pg_create_physical_replication_slot |        3 | false, false
 make_interval                       |        7 | 0, 0, 0, 0, 0, 0, 0.0
 jsonb_set                           |        4 | true
 pg_backup_start                     |        2 | false
 jsonb_path_exists                   |        4 | '{}'::jsonb, false
 jsonb_path_match                    |        4 | '{}'::jsonb, false
 jsonb_path_query_array              |        4 | '{}'::jsonb, false
 jsonb_path_query_first              |        4 | '{}'::jsonb, false
 jsonb_path_match_tz                 |        4 | '{}'::jsonb, false
 jsonb_path_query_tz                 |        4 | '{}'::jsonb, false
 jsonb_path_query_first_tz           |        4 | '{}'::jsonb, false
 normalize                           |        2 | 'NFC'::text
 pg_backup_stop                      |        1 | true
 pg_promote                          |        2 | true, 60
 install_extension                   |        5 | NULL::text[]
 concat_text                         |        2 | 'hello, hi'::text, 'ok, bye'::text
 get_json_obj                        |        2 | '{"key": "value::json"}'::json
 add_floats                          |        2 | ((1)::numeric + 3.14)
(33 rows)
```

We can probably parse const expressions but more complex default expressions will likely still fail.

Note that Postgraphile returns null for `defaultValue` for function args.

Reply:
> Note that Postgraphile returns null for defaultValue for function args.

Interesting, yeah, sounds like a hard problem to do perfectly.
It does seem like handling simple scalar cases like int/bigint/text/varchar/bool would make for easier to understand APIs but its a backwards compatible change to show the default value

does that scope seem more reasonable?

Reply:
I'll be OOO so will approve this. Up to you if you'd like to handle those scalar cases here or merge it & add that to https://github.com/supabase/pg_graphql/issues/410

Reply:
Implemented a simple algo to parse some basic types on a best effort basis. This can be made more sophisticated over time.